api/endpoint.go

```diff
@@ -276,8 +276,17 @@ func (request AttributeBetweenSurfacesRequest) execute(
 }
 
 func parseGetRequest(ctx *gin.Context, v Normalizable) error {
-	if err := json.Unmarshal([]byte(ctx.Query("query")), v); err != nil {
-		return core.NewInvalidArgument(err.Error())
+	query, status := ctx.GetQuery("query")
+	if (!status){
+		return core.NewInvalidArgument(
+			"The specified endpoint requires a 'query' parameter  ",
+		)
+	}
+	if err := json.Unmarshal([]byte(query), v); err != nil {
+		msg := "Please ensure that the supplied query is valid " +
+			"and conforms to the expected swagger Request specification: "
+		return core.NewInvalidArgument(
+			fmt.Sprintf(msg+" %v", err.Error()))
```

Comment:
Is the "fmt.Sprintf(msg+" %v", err.Error()))" for debugging or do we expect somebody to read the standard out on the server?

Reply:
Yes, If an error occurs in this code, it will be logged on the server terminal just like any other error. In the production environment, we may want to hide this error from the server terminal. In that case, we need to modify the error handling in the code and add a proper logging mechanism for debugging in the production environment.

Reply:
Not only are those logged in to production server, they are forwarded to the user.
We also want to know where our users screw up! Helps with audit and problem-solving.

So these are proper production messages, no debugging.

Reply:
Ok, but I guess we want to turn off the output to terminal in the production release? Terminal messages are very slow and no-one will ever see the message.

Reply:
No, we do not want to turn them off.

They are very useful for us. By having them on output we ourselves can see what user did wrong, without relying on the client code (maybe they swallow the error message or handle it incorrectly). We can also spot weird behavior, aka someone trying to access unlisted storage accounts or constantly submitting wrong sas tokens.

And there is no point for special handling of this particular error message. Anyway we are logging all the requests on server console. If this is an error, I don't care about additional slowness of one more info.
Honest user errors won't happen that often.